src/main/java/com/github/manolo8/darkbot/BackpageManager.java

```
@@ -67,11 +70,40 @@ private int validTick() {
     }
 
     private int sidKeepAlive() throws Exception {
-        HttpURLConnection conn = (HttpURLConnection) new URL(instance + "indexInternal.es?action=" + getRandomAction())
-                .openConnection();
-        conn.setInstanceFollowRedirects(false);
-        conn.setRequestProperty("Cookie", "dosid=" + sid);
-        return conn.getResponseCode();
+        return getConnection("indexInternal.es?action=" + getRandomAction()).getResponseCode();
+    }
+
+    public HttpURLConnection getConnection(String params) throws Exception{
+        if (!isInvalid()) {
+            HttpURLConnection conn = (HttpURLConnection) new URL(this.instance + params)
+                    .openConnection();
+            conn.setInstanceFollowRedirects(false);
+            conn.setRequestProperty("Cookie", "dosid=" + this.sid);
+            return conn;
+        }
+        return null;
```

It really makes no sense to return a null connection. Either you can make a connection, or the method should throw an exception. I'd rather get an UnsupportedOperationException than a NullPointerException for assuming this isn't null.